src/raytracer/qdGenerator.m

```diff
@@ -181,4 +229,79 @@
 % Wrap azimuth to [0,360)
 az = mod(az, 360);
 
+end
+
+
+function output = getTgayQdOutput(dRayOutput, scenarioName, prePostParam)
+intraClusterParams = getIntraClusterParams(scenarioName, prePostParam);
+aodAzCursor = dRayOutput(10);
+aodElCursor = dRayOutput(11);
+aoaAzCursor = dRayOutput(12);
+aoaElCursor = dRayOutput(13);  
+
+if intraClusterParams.n ~= 0
+    output = zeros(intraClusterParams.n,21);
+    % generate delays
+    taus = nan(intraClusterParams.n+1, 1);
+    taus(1) = dRayOutput(8); 
+    i = 1;
+    while i <= intraClusterParams.n
+        diff = randomExponetialGenerator(intraClusterParams.lambda);
+        taus(i+1) = taus(i)+intraClusterParams.delayMultiplier*diff;
+        output(i,8) = taus(i+1);
+        i = i+1;
+    end
+    
+    % generate path gain, doa/dod, AOA/AOD (AZ & EL) and phase offset 
+    for i = 1:intraClusterParams.n
```

Comment:
Please keep delays outside otherwise it would be really tough to understand for the user (in case they want to change for new model) 
Also, I dont want to change this at this stage.  We dont have time for testing again.